src/common/algorithm_scale.hpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file algorithm_scale.hpp
+ * @author Radek Vana
+ * @brief template function set to scale numbers
+ * originally used template <class OUT_TYPE, class IN_TYPE>
+ * but I would need to know which type is bigger to use it to calculations
+ * @date 2021-10-15
+ */
+
+#pragma once
+#include <algorithm>
+
+/**
+ * @brief scale number from original to scaled range
+ *
+ * @tparam T
+ * @param num number to be scaled
+ * @param min original minimum range
+ * @param max original maximum range
+ * @param scaled_min scaled minimum range
+ * @param scaled_max scaled maximum range
+ * @return T constexpr
+ */
+template <class T>
+T constexpr scale(T num, T min, T max, T scaled_min, T scaled_max) {
+    if (scaled_min == scaled_max)
+        return scaled_min;
+    if (min == max)
+        return scaled_max;
+
+    if (min > max) {
+        std::swap(min, max);
+        std::swap(scaled_min, scaled_max);
+    }
+
+    num = std::clamp(num, min, max);
+
+    if (scaled_min > scaled_max) {
+        std::swap(scaled_min, scaled_max);
+        num = num - min;
+        num = max - num;
+    }
+    return (scaled_max - scaled_min) * (num - min) / (max - min) + scaled_min;
```

Comment:
Can be dangerous in `int` arithmetic because of division. At least, should be stated in comments that integers will give different results from floats.

Reply:
max == min is checked
and it will first multiply, than divide to not loose precision